parachain/pallets/ethereum-light-client/src/tests.rs

```diff
@@ -719,7 +719,62 @@ fn it_should_not_allow_handle_long_range_fork_for_blocks_that_cannot_be_finalize
 }
 
 #[test]
-fn it_should_not_allow_handle_long_range_fork_for_blocks_at_or_after_current_finalized() {
+fn it_should_not_allow_force_reset_to_fork_for_blocks_at_or_after_current_finalized() {
+	new_tester::<Test>().execute_with(|| {
+		let block1 = child_of_genesis_ethereum_header();
+		let block1_hash = block1.compute_hash();
+		let block2 = child_of_header(&block1, Some(1));
+		let block2_hash = block2.compute_hash();
+		let block3 = child_of_header(&block2, Some(2));
+		let block3_hash = block3.compute_hash();
+		let block4 = child_of_header(&block3, Some(3));
+		let block4_hash = block4.compute_hash();
+
+		let bad_block2 = child_of_header(&block1, Some(4));
+		let bad_block2_hash = block2.compute_hash();
+
+		// Initial chain:
+		// Time   Block
+		// 0      B0
+		//        |
+		// 1      B1
+		//        |
+		// 2      B2      Finalized Block
+		//        |    
+		// 3      B3   
+		//        |    
+		// 4      B4      Best Block
+		let ferdie: AccountId = Keyring::Ferdie.into();
+		for header in vec![block1.clone(), block2, block3, block4].into_iter() {
+			assert_ok!(Verifier::import_header(
+				Origin::signed(ferdie.clone()),
+				header,
+				Default::default(),
+			));
+		}
+
+		// Relies on DescendantsUntilFinalized = 2
+		assert_eq!(<FinalizedBlock<Test>>::get().hash, block2_hash);
+		assert!(<Headers<Test>>::get(block2_hash).unwrap().finalized);
+		assert!(<Headers<Test>>::get(block3_hash).unwrap().finalized == false);
+		assert!(<Headers<Test>>::get(block4_hash).unwrap().finalized == false);
+		assert_eq!(BestBlock::<Test>::get().0.hash, block4_hash);
+
+		assert_err!(
+			Verifier::force_reset_to_fork(Origin::root(), block2_hash),
+			DispatchError::Other("Cannot reset to fork after the current finalized block.")
+		);
+
+		assert_eq!(<FinalizedBlock<Test>>::get().hash, block2_hash);
+		assert!(<Headers<Test>>::get(block2_hash).unwrap().finalized);
+		assert!(<Headers<Test>>::get(block3_hash).unwrap().finalized == false);
+		assert!(<Headers<Test>>::get(block4_hash).unwrap().finalized == false);
+		assert_eq!(BestBlock::<Test>::get().0.hash, block4_hash);
+	});
+}
+
+#[test]
+fn it_should_not_allow_force_reset_to_fork_for_blocks_that_are_not_finalized() {
```

Comment:
Ditto. Could be shortened to:
```suggestion
fn should_fail_reset_for_blocks_that_are_not_finalized() {
```